components/sync15/src/clients/engine.rs

```diff
@@ -83,15 +97,26 @@ impl<'a> Driver<'a> {
                     self.memcache_max_record_payload_size(),
                 )?;
 
+                // Add the new client record to our map of recently synced
+                // clients, so that downstream consumers like synced tabs can
+                // access them.
+                self.recent_clients.insert(
```

Comment:
Nit: Can we factorize this and below before the `if` condition as we are cloning everything anyway?